### DISCUSSION TITLE
specific peak2gene link

### DISCUSSION THREAD

**Discussion by rojinsafavi** [GreenleafLab/ArchR#1134] [Questions / Documentation]
I wanted to see if there is a way to show a specific peak2gene correlation link in the browser? 

1. adding the peak2gene correlation
```
projHeme2 <- addPeak2GeneLinks(
    ArchRProj = projHeme2,
    reducedDims = "IterativeLSI"
)
```
2. retrieving the data frame

```
p2geneDF <- metadata(projHeme2@peakSet)$Peak2GeneLinks
p2geneDF$geneName <- mcols(metadata(p2geneDF)$geneSet)$name[p2geneDF$idxRNA]
p2geneDF$peakName <- (metadata(p2geneDF)$peakSet %>% {paste0(seqnames(.), "_", start(.), "_", end(.))})[p2geneDF$idxATAC]


p2geneDF = as.data.frame(p2geneDF)
p2geneDF = na.omit(p2geneDF)

S = strsplit(p2geneDF[,c('peakName')], '_')
S = as.data.frame(S)
S = t(S)
rownames(S) = seq(1, length(rownames(S)))

p2geneDF$chr   = S[,1]
p2geneDF$start = S[,2]
p2geneDF$end   = S[,3]

p2geneDF = p2geneDF[,c('chr', 'start', 'end', 'Correlation', 'FDR', 'VarQATAC', 'VarQRNA', 'geneName', 'peakName')]
```

```head(p2geneDF)```

<img width="668" alt="Screen Shot 2021-11-02 at 8 02 50 PM" src="https://user-images.githubusercontent.com/19545198/140004617-6c8e5a8c-701b-4878-8220-aa5a1a560e5e.png">

```
dim(p2geneDF)
16728109
```

Now I'd like to look at this specific peak

```
chr1_11843973_11844473
```

<img width="747" alt="Screen Shot 2021-11-02 at 8 08 45 PM" src="https://user-images.githubusercontent.com/19545198/140004977-d5e5ff67-a51e-401f-acf7-1890e7e00e33.png">

but the peak is not there when I return the loop?

<img width="617" alt="Screen Shot 2021-11-02 at 8 10 30 PM" src="https://user-images.githubusercontent.com/19545198/140005078-5deb7b8b-8700-49a5-9efa-46d544816759.png">

**Comment by rcorces** [comment]
I would imagine this is because you are using `returnLoops = TRUE` which, combined with `resolution` is going to alter the coordinates of the loops returned. So the precise coordinates do not match. Try it with `returnLoops = FALSE`

**Comment by rojinsafavi** [comment]
Thanks @rcorces  - I tried that before but it seems that the function requires a grange obj, for now, I am converting the data frame to the grange and using that - it seems to be working , but it appears that I'm geeting peak2peak visualization instead of peak2 gene? this is the comparison of using the loop vs making my own grange for specific peaks
<img width="369" alt="Screen Shot 2021-11-02 at 11 06 25 PM" src="https://user-images.githubusercontent.com/19545198/140016018-2ff33184-1772-4d19-b702-c3c031c8d054.png">
<img width="370" alt="Screen Shot 2021-11-02 at 11 06 35 PM" src="https://user-images.githubusercontent.com/19545198/140016029-9df1f199-c1a7-428f-a51e-67ab03df226e.png">

**Reply by rcorces** [reply]
I'm not sure I understand. You can try `returnLoops = TRUE` and `resolution = NULL`. Otherwise I think I need a reproducible example to help more.

There is another open discussion (https://github.com/GreenleafLab/ArchR/discussions/1127) which seems to be reporting something similar but I dont have a way to give specific advice without a reproducible example from the tutorial data
